junit-vintage-engine/src/testFixtures/java/org/junit/vintage/engine/samples/junit4/ConcurrencyTests.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2015-2020 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * https://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.vintage.engine.samples.junit4;
+
+import org.junit.Test;
+
+/**
+ * Vintage tests to reproduce various concurrency scenarios.
+ *
+ * @since 5.8
+ */
+public class ConcurrencyTests {
+
+	public static class A {
+		@Test
+		public final void test1() throws InterruptedException {
+			Thread.sleep(5);
```

Comment:
Please follow the strategy employed in `ParallelExecutionIntegrationTests` to make these tests faster.

Reply:
Looks like this is a holdover from a previous implementation of the tests. There is no need for `Thread.sleep` because the test listener is synchronized.

Reply:
I removed the calls to `Thread.sleep`.

Reply:
This seems to cause problems on some platforms (e.g. Windows and JDK 16). I'm investigating.

Reply:
I used the `CyclicBarrier` approach from `ParallelExecutionIntegrationTests` and the tests pass on all platforms now.

Reply:
Looks good! 👍